### ISSUE TITLE
New rule: padded-object

### ISSUE THREAD

**Issue by Aladdin-ADD** [eslint/eslint#9048]
**Please describe what the rule should do:**

seems like we have been lack of linting blank lines in multiline object properties.

**What category of rule is this? (place an "X" next to just one item)**

[x] stylistic

**Provide 2-3 code examples that this rule will warn about:**

```js
/* padded-object: ["error", "never"]*/
var foo = {

  bar: 0

}

/* padded-object: ["error", "both"]*/
var foo = {
  bar: 0
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

This rule seems like a fairly universal and common requirement. It meets all the core rule guidelines.

**Comment by Aladdin-ADD** [comment]
btw, I'm not sure about the rulename and options -- it should be consistent to the similar rules.

**Comment by not-an-aardvark** [comment]
I'm wondering if we should just add this into [`padded-blocks`](http://eslint.org/docs/rules/padded-blocks). (Yes, we're completely stretching the definition of what a "block" is, but the rule already does that by including class bodies and switch statements.)

**Comment by bwatson3** [comment]
Is there any update on this? From the example above I can't tell if this is just looking for a way to require/prohibit padding at the top and bottom of object blocks, but I'm looking for a way to require padding lines between each object member or preferably between members with their own blocks. For example, a rule that would allow this:
```javascript
const a = {
  b() {
    // some code
  },

  c() {
    // some more code
  },
};
```
but give an error message if you did this:
```javascript
const a = {
  b() {
    // some code
  },
  c() {
    // some code
  }
};
```
So that we could enforce this rule from Airbnb's javascript style guide https://github.com/airbnb/javascript#whitespace--after-blocks

**Comment by rwwagner90** [comment]
@bwatson3 I have the same question. My need is for enforcing lines after methods in objects.

**Comment by kaicataldo** [comment]
I don't think this should be rolled into `padded-blocks`, since that rule focuses on padded newlines at the beginning and end of "blocks" (as @not-an-aardvark says, we definitely stretch the definition of what a block is) - so, in other words, it's really `padded-newlines-in-blocks`.

I think it would make sense to make a new rule that's analogous to [lines-between-class-members](https://eslint.org/docs/rules/lines-between-class-members) that has similar configuration called `lines-between-object-properties`.

**Comment by rwwagner90** [comment]
@kaicataldo I agree that this should not be for padding, but for lines after methods in objects.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.

**Comment by rwwagner90** [comment]
@not-an-aardvark how are we supposed to get consensus? Are all issues guaranteed to be viewed by the team?

**Comment by not-an-aardvark** [comment]
Ideally, if an issue is open for long enough, it gets reviewed by the team. However, we get a lot of feature requests and we wouldn't realistically be able to implement and maintain all of them, so we err on the side of not accepting proposals after awhile unless there is a clear consensus in favor accepting them. In this case, there wasn't a strong consensus one way or the other, so I closed the issue.

If this is still something you'd like to advocate for, feel free to open a new issue.
